streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -28,29 +28,34 @@
     private final Set<TaskId> prevActiveTasks;
     private final Set<TaskId> prevAssignedTasks;
 
+    private String rackId;
     private int capacity;
 
-
     public ClientState() {
         this(0);
     }
 
-    ClientState(final int capacity) {
-        this(new HashSet<TaskId>(), new HashSet<TaskId>(), new HashSet<TaskId>(), new HashSet<TaskId>(), new HashSet<TaskId>(), capacity);
+    public ClientState(int capacity) {
+        this(0, "");
+    }
+
+    ClientState(final int capacity, final String rackId) {
+        this(new HashSet<TaskId>(), new HashSet<TaskId>(), new HashSet<TaskId>(), new HashSet<TaskId>(), new HashSet<TaskId>(), capacity, rackId);
     }
 
-    private ClientState(Set<TaskId> activeTasks, Set<TaskId> standbyTasks, Set<TaskId> assignedTasks, Set<TaskId> prevActiveTasks, Set<TaskId> prevAssignedTasks, int capacity) {
+    private ClientState(Set<TaskId> activeTasks, Set<TaskId> standbyTasks, Set<TaskId> assignedTasks, Set<TaskId> prevActiveTasks, Set<TaskId> prevAssignedTasks, int capacity, String rackId) {
```

Comment:
Nit: Can we break this long line into one line per parameter?

also, please add `final` to each parameter. Thx.